src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -253,27 +282,39 @@ def create_empty_extra_products(self):
   def select(self, target):
     if not isinstance(target, JvmTarget):
       return False
-    if self._classify_compile_target(target) is not None:
-      return True
-    return False
+    return target.has_sources('.java') or target.has_sources('.scala')
 
   def _classify_compile_target(self, target):
-    return _JvmCompileWorkflowType.classify_target(target)
+    if not target.has_sources('.java') and not target.has_sources('.scala'):
+      return None
+
+    if not self._ignore_tags_when_calculating_workflow:
+      for t in target.tags:
+        flow = self._compiler_tags.get(t)
+        if flow:
+          return _JvmCompileWorkflowType(flow)
+
+    if self._default_workflow == 'zinc-only':
```

Comment:
Post #7304, this could be an enum as well, which would simplify choices selection and allow `.resolve_for_enum_variant()` here, which avoids the manual `ValueError`.

Reply:
yes. I think that would be a good simplification.